packages/blade/src/components/Dropdown/_decisions/controlled-dropdown.md

```diff
@@ -0,0 +1,192 @@
+  <SelectInput />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Bangalore" value="bangalore" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>;
+```
+
+Inspirations
+
+</details>
+
+### Approach Comparison
+
+|                          | ✅ `onChange` + `isSelected`                                                                                                                                                                                                                                | `onChange` + `value`                                                                                                                                                                                                                                                                                                                                    | `useSelect`                                                                                                                                                                                                                                                              |
+| ------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| State Management         | ✅ Independent of how state is manged (consumers can store value of item in state as shown in example above, they can store selectedIndex, they might also have entire ActionList data as array somewhere and they can use `selected` prop from their data) | Consumer can store `currentSelectionValue` in state and handle on basis of that                                                                                                                                                                                                                                                                         | Handles state internally. Requires FE Core to export appropriate methods. E.g. If consumer is storing selected data in some object, they have to transform that data to our input format and then call `selectItem`                                                      |
```

Comment:
So in `isSelected` approach, your selection logic is part of JSX so it can be written in rendering part which would be more intuitive when your data is stored in different ways.

Whereas with something like `selectItem()` method, you have to call that function when state is updating to select an item.

You can checkout example in this section - https://github.com/razorpay/blade/blob/saurabh/controlled-dropdown/api-decisions/packages/blade/src/components/Dropdown/_decisions/controlled-dropdown.md#why-not-useselect